lisa/tools/virtualclient.py

```diff
@@ -0,0 +1,275 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import csv
+import json
+from datetime import datetime
+from decimal import Decimal
+from pathlib import Path
+from typing import Any, List, Tuple
+
+from lisa import notifier
+from lisa.base_tools import Uname, Wget
+from lisa.executable import Tool
+from lisa.messages import VCMetrics
+from lisa.operating_system import Debian
+from lisa.util import LisaException
+from lisa.util.process import Process
+
+from .chmod import Chmod
+from .chown import Chown
+from .git import Git
+from .ln import Ln
+from .parted import Parted
+from .unzip import Unzip
+from .whoami import Whoami
+
+
+class VirtualClientTool(Tool):
+    _version = "1.14.36"
+
+    @property
+    def command(self) -> str:
+        return "virtualclient"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        uname = self.node.tools[Uname]
+        arch = uname.get_linux_information().hardware_platform
+        if arch == "x86_64":
+            arch = "x64"
+        elif arch == "aarch64":
+            arch = "arm64"
+        else:
+            raise LisaException(f"not supported architecture {arch}")
+
+        self._vc_download_path: str = (
+            f"{self.node.working_path.parent.parent}/virtualclient"
+        )
+        self._vc_raw_tool_path: str = (
+            f"{self._vc_download_path}/content/linux-{arch}/VirtualClient"
+        )
+        self._vc_profile_file_path: str = (
+            f"{self._vc_download_path}/content/linux-{arch}/profiles"
+        )
+        self._vc_log_path: str = f"{self._vc_download_path}/content/linux-{arch}/logs"
+
+    def install(self) -> bool:
+        if isinstance(self.node.os, Debian):
+            _ = self.node.tools[Git]
+            _ = self.node.tools[Parted]
+            self.node.os.install_packages(["gnupg", "lshw"])
+        vc_download_url = (
+            f"https://www.nuget.org/api/v2/package/VirtualClient/{self._version}"
+        )
+        package_name = f"virtualclient.{self._version}.nupkg"
+        wget_tool = self.node.tools[Wget]
+        wget_tool.get(
+            vc_download_url, str(self.node.working_path.parent.parent), package_name
+        )
+
+        unzip = self.node.tools[Unzip]
+        unzip.extract(
+            f"{self.node.working_path.parent.parent}/{package_name}",
+            self._vc_download_path,
+        )
+
+        self.node.tools[Chmod].chmod(self._vc_raw_tool_path, "777")
+        ln = self.node.tools[Ln]
+        ln.create_link(self._vc_raw_tool_path, "/usr/bin/virtualclient")
+        return self._check_exists()
+
+    def generate_layout(self, clients_info: List[Tuple[str, str, str]]) -> str:
```

Comment:
Use named parameters, not a Tuple. It's not self explained.

Reply:
updated